frontend/src/App.tsx

```diff
@@ -394,13 +400,31 @@ class App extends PureComponent<Props, State> {
       // how many projects are being created with Streamlit while still keeping
       // possibly-sensitive info like the scriptPath outside of our metrics
       // services.
-      reportHash: hashString(SessionInfo.current.installationId + scriptPath),
+      reportHash: newReportHash,
     })
 
-    this.setState({
-      reportId: newReportProto.id,
-      reportName: name,
-    })
+    if (reportHash !== newReportHash) {
```

Comment:
Flip this `if` condition to `if (reportHash === newReportHash)` for readability (and also flip the if/else branches to match, of course!)

Otherwise, the `else` branch is a double-negative, which is hard to reason about.

So instead of:
```ts
if (x !== y) {
  foo
} else {
  // This branch is hit when x is not not equal to y. That's a double-negative!
  bar
} 
```

do 

```ts
if (x === y) {
  bar
} else {
  foo
}
```